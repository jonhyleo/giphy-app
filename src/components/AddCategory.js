import { useState } from "react";
import PropTypes from "prop-types";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTimesCircle } from "@fortawesome/free-solid-svg-icons";

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");
  const [error, setError] = useState(false);

  const handleInputChange = (e) => {
    setError(false);
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((category) => [inputValue, ...category]);
      setInputValue("");
    } else {
      setError(true);
    }
  };

  return (
    <>
      <form onSubmit={handleSubmit} className="mb-2">
        <div className="form-group">
          <input
            type="text"
            className="form-control"
            placeholder="Type a category"
            onChange={handleInputChange}
          />
        </div>
      </form>
      {error ? (
        <div
          className="alert alert-danger alert-dismissible fade show"
          role="alert"
        >
          <FontAwesomeIcon icon={faTimesCircle} size="lg" fixedWidth />
          The field must contain minimum 3 letters and cannot be empty!
        </div>
      ) : (
        ""
      )}
    </>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
