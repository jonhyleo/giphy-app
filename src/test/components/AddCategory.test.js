import React from "react";
import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe("Test on AddCategory component", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("Component should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Text input should change", () => {
    const input = wrapper.find("input");
    const value = "Hola mundo";

    input.simulate("change", {
      target: {
        value,
      },
    });
  });

  test("shouldn't post info with submit", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("should post info with submit", () => {
    const input = wrapper.find("input");
    const value = "Hola mundo";

    // Simulate inputChange
    input.simulate("change", {
      target: {
        value,
      },
    });

    // Simulate submit event
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    // Call setCategories
    expect(setCategories).toHaveBeenCalled();

    // Clear input value
    expect(input.prop("value")).toBe("");
  });
});
