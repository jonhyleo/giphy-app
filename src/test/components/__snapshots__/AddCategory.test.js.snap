// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on AddCategory component Component should render properly 1`] = `
<Fragment>
  <form
    className="mb-2"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Type a category"
        type="text"
        value=""
      />
    </div>
  </form>
</Fragment>
`;
